### PR TITLE
Refactor: Make blocks location-aware

### PR DESCRIPTION
**Pull request by greg-does-weather**
## What does this PR do? 🛠️

Instead of pushing route controllers into the data service and having it figure out the location to work with, have blocks do that. This PR adds a `getLocation` method to the block base class that we can build on in the future to allow more different ways of specifying location. It also updates the weather data service `getCurrentConditions` method to `getCurrentConditionsFromGrid` to accept the WFO grid as arguments instead of deriving it from the route.

## What does the reviewer need to know? 🤔

I will do the other weather data service methods in a follow-on PR.

### CODE REVIEW

File: web/modules/weather_blocks/src/Plugin/Block/WeatherBlockBase.php [hunk 1]
```diff
@@ -67,4 +67,31 @@ public function getCacheMaxAge() {
     return 0;
   }
 
+  /**
+   * Get the current location that the block should be associated with.
+   *
+   * If the current route is on the grid, pulls the WFO grid information from
+   * the URL.
+   */
```
Comment (eric-gade) [inline web/modules/weather_blocks/src/Plugin/Block/WeatherBlockBase.php:75]: Leave or toss, but I was thinking some pointers in the comment might help us keep track of the structure of this location object as we modify it going forward:
```suggestion
  /**
   * Get the current location that the block should be associated with.
   *
   * If the current route is on the grid, pulls the WFO grid information from
   * the URL.
   * @returns array $location An array with location info in the following
   * structure:
   *     array['grid'] An NWS API grid response
   *.                       ['wfo'] The NWS WFO code for the location
   *                       ['gridX'] The X value of the grid coordinate
   *                       ['gridY'] The Y value of the grid coordinate
   */
```

File: web/modules/weather_data/src/Service/WeatherDataService.php [hunk 2]
```diff
@@ -268,7 +250,7 @@ public function getCurrentConditions($route) {
         'feels_like' => (int) round($feelsLike),
         'humidity' => (int) round($obs->relativeHumidity->value ?? 0),
         'icon' => $this->legacyMapping->$obsKey->icon,
-        'location' => $location,
+        'location' => "SOMEWHERE - FIX THIS SOON",
```
Comment (eric-gade) [inline web/modules/weather_data/src/Service/WeatherDataService.php:253]: I assume related to https://github.com/weather-gov/weather.gov/issues/474

**Review by eric-gade — APPROVED** [review]
Checks out on my end! 👍 
  
## Code Review Checklist

This is an automated comment on every pull request requiring a review. A checked item is either an assertion that I have tested this item or an indication that I have verified it does not apply to this pull request.

## The Basics
- [x] Checks are passing
- [x] I read the code
- [x] I ran the code
- [x] (if applicable) Post deploy steps are run
- [x] (if applicable) I validated the change on the deployed version in <environment>

## Documentation
- [x] changes to “how we do things” are documented in READMEs
- [x] all new functions and methods are commented using plain language
- [x] any new modules added documented in modules.md

## Security
- [x] security false positives are documented
- [x] data from external sources is cleaned and clearly marked

## Reliability
- [x] error handling exists for unusual or missing values
- [x] interactions with external systems are wrapped in try/except
- [x] functionality is tested with unit or integration tests
- [x] dependency updates in composer.json also got changed in composer-lock.json

## Infrastructure
- [x] all changes are auditable and documented via a script
- [x] it is clear who can and should run the script
- [x] (if applicable) diagrams have been updated or added in PlantUML

## Accessibility
- [x] New pages have been added to cypress-axe file so that they will be tested with our automated accessibility testing
- [x] Meets WCAG 2.0 AA or 2.1 AA for Section 508 compliance
    - [x] Site is keyboard accessible. All interactions can be accessed with a keyboard
    - [x] Site is free of keyboard traps. The keyboard focus is never trapped in a loop
    - [x] All form inputs have explicit labels
    - [x] Form instructions are associated with inputs
    - [x] All relevant images use an img tag
    - [x] All images have appropriate alt attributes
    - [x] Multimedia is tagged. All multimedia has appropriate captioning and audio description
    - [x] Text has sufficient color contrast. All text has a contrast ratio of 4.5:1 with the background
    - [x] Site never loses focus. Focus is always visible when moving through the page with the keyboard
    - [x] Tab order is logical
    - [x] Tables are coded properly. Tables have proper headers and column attributes
    - [x] Headings are nested properly. Heading elements are nested in a logical way
    - [x] Language is set. The language for the page is set
    - [x] CSS is not required to use the page. The page makes sense with or without CSS
    - [x] Links are unique and contextual. All links can be understood taken alone, e.g., ‘Read more - about 508’
    - [x] Page titles are descriptive

## Device Matrix
- [x] firefox/gecko (renders correctly and user interactions work)
- [x] chrome/chromium/edge (renders correctly and user interactions work)
- [x] safari/webkit (renders correctly and user interactions work)
- [x] web page is readable and usable 
    - [x] at 480px     (mobile)
    - [x] at 640px     (tablet)
    - [x] at 1024px    (desktop)

File: web/modules/weather_blocks/src/Plugin/Block/WeatherBlockBase.php (hunk 1 above)
Comment (greg-does-weather) [reply web/modules/weather_blocks/src/Plugin/Block/WeatherBlockBase.php:75]: Good call! I was also wondering if we should create some explicit types for these sorts of things (this location, current conditions, hourly forecast, etc.). That might be overkill, though, when we can just document them at the points they're created and they're each only created in one place. 🤔

**Pull request merged**